CONTRIBUTING.md

```diff
@@ -1,7 +1,175 @@
+
+	2. Now that you have an online ('remote') clone of `jwst`, it needs to be downloaded to your machine. To create this 'local' clone, choose an area on your file system and use the `git clone` command to clone your remote fork on to your machine: 
+		>> cd <directory>
+		>> git clone git@github.com:<your_username>/jwst.git`
+
+	3. Make sure that your references to 'origin' and 'upstream' are set correctly - you will need this to keep everything in sync and push your changes online. While your inital local clone will be an exact copy of your remote, which is an exact copy of the 'upstream' `spacetelescope/jwst`, these all must be kept in sync manually (via git fetch/pull/push). To check the current status of these references:
+		>> git remote -v
+
+	After your inital clone, you will likely be missing the reference to 'upstream' (which is just the most commonly used name in git to refer to the main project repository - you can call this whatever you want but the origin/upstream conventions are most commonly used) - to set this, use the `add` git command
+
+		If you are using an SSH key to authenticate.
+			>> git remote add upstream git@github.com:spacetelescope/jwst.git
+		Otherwise, you can simply set it to the repository URL but you will have to enter your password every time you fetch/push
+			>> git remote add upstream https://github.com/spacetelescope/jwst
+
+		If you ever want to reset these URLs, add references to other remote forks of `jwst` for collaboration, or change from URL to SSH, you can use the related `git remote set-url` command.
+
+### Step 2: Creating a branch for your changes
+
+It is a standard practice in git to create a new 'branch' (off 'master') for each new feature or bug fix. We will be creating a branch called 'my_feature' off of our local clone of the forked `jwst` repository:
+
+	1. Get your local 'master' branch up to date with any new changes upstream.
+		First, verify that you are currently on the `master` branch:
+			>> git status
+		and if your current branch isn't master,
+			>> git checkout master
+		Next, it is good practice to keep your local 'master' up-to-date with 'upstream/master'. To do this:
+			>> git pull upstream master
+		Note that `git pull` is sufficient here assuming you never make commits directly to 'master' - otherwise, you may have conflicts and this can get tricky. For this reason, it is recommended that you don't commit directly to master but rather branch off of it.
+
+	2. Now, create a new branch for your feature off your now up-to-date 'master' branch for your feature. You can call this branch whatever you like - in this example, we'll call it 'my_feature'
+
+		>> git checkout -b my_feature upstream/master
+
+		Doing this will create a new branch off local/master called 'my_feature', and automatically switch you over to your new branch.
```

Comment:
This is not quite correct. This creates a new branch off of upstream/master that is currently on disk. "local/master" is the "origin/master" on your disk.